Add integration test

jayay opened the pull request:
This integration test spawns a little HTTP server with SSL and another subprocess with PHP, controlled by `faketime`, which fakes the current time for the PHP process it has control over.

In that way, we have the possibility to create reproducible requests, and to monitor behavioral changes in the requests sent to our API. The sent request can be observed and tested for certain traits in the unit test, or be compared as a whole in the way it is right now.

**known bugs**
~If the PHP subprocess takes longer than one second to send the request, the timestamp might be off, also leading to a different HMAC.~ Fixed in 1a58577

jayay commented on tests/SDKIntegrationTest.php:
```diff
@@ -0,0 +1,104 @@
+            'datestart' => '25-03-2022 02:17:53',
+            'showcancelled' => true
+        ]);
+        $sdk->sendRequests('testtoken', 'testsecrect');
```
oh no, a typo

jayay commented on tests/SDKIntegrationTest.php:
```diff
@@ -0,0 +1,104 @@
+     */
+    public function test_request_structure_correct()
+    {
+        $ncat = new Process(['sudo', 'ncat', '-n', '--ssl', '-l', '1234', '-i', '1', '-4']);
```
todo to self: Port `1234` is not well-known, `sudo` might not be required here

dkaisers reviewed (approved):
LGTM

legionth reviewed (changes requested):
In general I think this is a very good idea. But this would also fail on most systems which have no local installation of the API-Server or wouldn't it?

Maybe this would be a good opportunity to add some additional [testsuits](https://phpunit.readthedocs.io/en/9.5/organizing-tests.html).

jayay commented:
@legionth I added a new test suite and renamed the old one. I did not feel the necessity to create new directories for those, I hope that's okay for now.

legionth reviewed (approved):
LGTM 👍
